rotkehlchen/chain/ethereum/modules/juicebox/decoder.py

```diff
@@ -0,0 +1,84 @@
+import logging
+from typing import Any
+
+from rotkehlchen.chain.ethereum.abi import decode_event_data_abi_str
+from rotkehlchen.chain.ethereum.modules.juicebox.constants import (
+    CPT_JUICEBOX,
+    PAY_ABI,
+    PAY_SIGNATURE,
+    TERMINAL_3_1_2,
+)
+from rotkehlchen.chain.ethereum.utils import token_normalized_value_decimals
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.constants.assets import A_ETH
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class JuiceboxDecoder(DecoderInterface):
+
+    def _decode_pay(self, context: DecoderContext) -> DecodingOutput:
+        """Decode pay with rewards in juicebox"""
+        if context.tx_log.topics[0] != PAY_SIGNATURE:
+            return DEFAULT_DECODING_OUTPUT
+
+        try:
+            _, decoded_data = decode_event_data_abi_str(context.tx_log, PAY_ABI)
+        except DeserializationError as e:
+            log.error(
+                f'Failed to deserialize Juicebox event at '
+                f'{context.transaction.tx_hash.hex()} due to {e}',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        amount = token_normalized_value_decimals(
+            token_amount=decoded_data[2],
+            token_decimals=18,
```

Comment:
why 18? Which token is it always? Can you add a comment here? Guess it's always ETH?